src/main/java/org/jenkinsci/plugins/codesonar/services/AlertsService.java

```diff
@@ -0,0 +1,110 @@
+package org.jenkinsci.plugins.codesonar.services;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.text.MessageFormat;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.http.client.utils.URIBuilder;
+import org.jenkinsci.plugins.codesonar.CodeSonarPluginException;
+import org.jenkinsci.plugins.codesonar.ResponseErrorException;
+import org.jenkinsci.plugins.codesonar.api.CodeSonarAlertCounter;
+import org.jenkinsci.plugins.codesonar.models.json.CodeSonarAlertData;
+
+import com.google.common.base.Throwables;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+public class AlertsService {
+    private static final Logger LOGGER = Logger.getLogger(AlertsService.class.getName());
+    
+    final private HttpService httpService;
+
+    public AlertsService(HttpService httpService) {
+        this.httpService = httpService;
+    }
+    
+    public CodeSonarAlertCounter getAlertCounter(URI baseHubUri, long analysisId) throws IOException {
+        CodeSonarAlertCounter counter = new CodeSonarAlertCounter();
+        //Loop through all possible kind values, currently from 0 to 26. Upper bound 50 is for future alert kinds.
+        for(long kind = 0; kind < 50; kind++) {
+            //Build char_table request URL
+            URIBuilder uriBuilder = new URIBuilder(baseHubUri);
+            uriBuilder.setPath(String.format("/analysis/%d-alert%d.json", analysisId, kind));
+            
+            URI requestUri = null;
+            try {
+                requestUri = uriBuilder.build();
+            } catch (URISyntaxException e) {
+                LOGGER.log(Level.WARNING, "Request URI for Alerts API contains a syntax error. %nException: {0}%nStack Trace: {1}", new Object[] {e.getMessage(), Throwables.getStackTraceAsString(e)});
+                return null;
+            }
+            
+            LOGGER.log(Level.INFO, "Querying alerts with kind={0}", kind);
+            InputStream jsonContent = null;
+            try {
+                jsonContent = httpService.getContentFromUrlAsInputStream(requestUri.toASCIIString());
+            } catch(ResponseErrorException e) {
+                if(e.getStatus() == 404) {
+                    /*
+                     * A 404 response is returned anytime the request specifies an alert kind that has no
+                     * occurrences on the specified analysis
+                     */
+                    LOGGER.log(Level.INFO, "Skipping current alert kind", e);
+                    continue;
```

Comment:
> "Skipping current alert kind"

I think that without context, this log message will not make sense.  I think you should reword it to say more-or-less what your code comment says:

> "HTTP 404: no alerts found for kind={0}"